lib/reesmat.gd

```diff
@@ -81,10 +81,8 @@ DeclareOperation( "ReesZeroMatrixSubsemigroup",
   [IsReesZeroMatrixSubsemigroup, IsList, IsSemigroup, IsList]);
 DeclareGlobalFunction("ReesZeroMatrixSubsemigroupNC");
 
-DeclareAttribute("Matrix", IsReesMatrixSubsemigroup);
-DeclareAttribute("Matrix", IsReesZeroMatrixSubsemigroup);
-DeclareSynonymAttr("MatrixOfReesMatrixSemigroup", Matrix);
-DeclareSynonymAttr("MatrixOfReesZeroMatrixSemigroup", Matrix);
+DeclareAttribute("MatrixOfReesMatrixSemigroup", IsReesMatrixSubsemigroup);
+DeclareAttribute("MatrixOfReesZeroMatrixSemigroup", IsReesZeroMatrixSubsemigroup);
```

Comment:
Up to now, there was just one attribute `Matrix` with two synonyms `MatrixOfReesMatrixSemigroup`, `MatrixOfReesZeroMatrixSemigroup`.
Do we really need/want two independent attributes `MatrixOfReesMatrixSemigroup`, `MatrixOfReesZeroMatrixSemigroup`?

Reply:
What would be an alternative way of doing this, so that there were not two independent attributes? We could have one attribute called something like `MatrixOfReesOrReesZeroMatrixSemigroup` with the other names being synonyms for it, but that feels a bit weird to me.

Reply:
Since there was just one attribute with different names `MatrixOfReesMatrixSemigroup`, `MatrixOfReesZeroMatrixSemigroup` up to now, the situation seems to be that either no object lies in both `IsReesMatrixSubsemigroup` and `IsReesZeroMatrixSubsemigroup` or the attribute values for objects in both filters are equal.
In the former case, introducing two independent attributes is on the one hand safer (one gets an error if one calls the wrong attribute), but on the other hand `Matrix` methods for the two cases make the distinction disappear, and from this point of view just one filter is wasted.
In the latter case, introducing two independent attributes may cause that the same value has to be computed twice, if one asks for the values of both attributes.

Reply:
The former case is the true one: `IsReesMatrixSemigroupElement` and `IsReesZeroMatrixSemigroupElement` (from which `IsReesMatrixSubsemigroup` and `IsReesZeroMatrixSubsemigroup` are derived) are different categories, and so a GAP object cannot lie in both.

Is it a problem to 'waste' filters?

Reply:
I asked @wilfwilson beforehand to make sure that the filters are mutually exclusive, and hence split this into two attributes. The alternative would IMHO not be to introduce `MatrixOfReesOrReesZeroMatrixSemigroup`, but rather I'd just turn one `DeclareAttribute` back into `DeclareSynonymAttr`.

But I really think it's best the way I now did it. Yes, we loose one filter bit. I don't see an issue with that, we have plenty bits left ;-).